src/sale.rs

```diff
@@ -804,6 +804,18 @@ impl Contract {
         sale.distribute_token_id = Some(distribute_token_id);
         self.sales.insert(&sale_id, &VSale::Current(sale));
     }
+    
+    #[private]
+    pub fn update_sale_price(&mut self, sale_id: u64, price: u128) {
```

Comment:
Please consider U128 type for price, it will allow to send values using JS as strings